DietPi-Software | PiVPN: Add unattended install

bastianpaetzold opened the pull request:
This is a possible implementation of unattended install of PiVPN (#3350)

I know there are probably still open points like if  `/boot/unattended_pivpn.conf` should be renamed or deleted after the installation, but I hope this can serve as a starting point. Please let me know if I should adjust anything.

I tested the installation with and without the `/boot/unattended_pivpn.conf` present as well as reinstall after an unattended install.

bastianpaetzold committed:
added unattended install support for pivpn

bastianpaetzold committed:
fixed unattended option being ignored for pivpn

bastianpaetzold committed:
simplified unattended pivpn install

removed variable to configure pivpn option file
added renaming of option file after install

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8117,12 +8118,20 @@ _EOF_
+
+			if [[ -f /boot/unattended_pivpn.conf ]]; then
+				G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' /boot/unattended_pivpn.conf
+				install_options='--unattended /boot/unattended_pivpn.conf'
```
```suggestion
				install_options=('--unattended' '/boot/unattended_pivpn.conf')
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8117,12 +8118,20 @@ _EOF_
 			Configure_iptables
 
-			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash || aSOFTWARE_INSTALL_STATE[$software_id]=0
+			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash $install_options || aSOFTWARE_INSTALL_STATE[$software_id]=0
```
Shellcheck doesn't like it when non-explicit variable splitting is done, hence we use arrays in such cases:
```suggestion
			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash "${install_options[@]}" || aSOFTWARE_INSTALL_STATE[$software_id]=0
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8117,12 +8118,20 @@ _EOF_
-			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash || aSOFTWARE_INSTALL_STATE[$software_id]=0
+			G_EXEC_OUTPUT=1 G_EXEC_NOEXIT=1 G_EXEC ./install.bash $install_options || aSOFTWARE_INSTALL_STATE[$software_id]=0
 			G_EXEC_NOHALT=1 G_EXEC rm install.bash
+			[[ -f /boot/unattended_pivpn.conf ]] && G_EXEC_NOHALT=1 G_EXEC mv /boot/unattended_pivpn.conf{,.ignore}
```
To be more clear?
```suggestion
			[[ -f /boot/unattended_pivpn.conf ]] && G_EXEC_NOHALT=1 G_EXEC mv /boot/unattended_pivpn.conf{,.applied}
```

bastianpaetzold replied:
Yes, way better

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8117,12 +8118,20 @@ _EOF_
+			local install_options
+
+			if [[ -f /boot/unattended_pivpn.conf ]]; then
+				G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' /boot/unattended_pivpn.conf
```
What is this setting used for? Generally, if users set it to something other than `0`, shouldn't we respect that? Generally, we have some nice function to apply settings transparently 🙂:
```suggestion
				G_CONFIG_INJECT 'UNATTUPG=[[:digit:]]' 'UNATTUPG=0' /boot/unattended_pivpn.conf
```

bastianpaetzold replied:
I totally agree and I normally wouldn't add something like this, but we already have this for interactive install: `G_EXEC sed -i 's/^[[:blank:]]*UNATTUPG=[[:digit:]]/UNATTUPG=0/' install.bash` causes `UNATTUPG=0` to be set regardless of the users input (if I understand it correctly). 
So I would prefer to respect the users choice in either case but didn't want to have inconsistent behavior between interactive and unattended install.

MichaIng replied:
Ah, it's about the `unattended-upgrades` package, not about PiVPN install/upgrades being unattended. Yes then it makes sense 👍. You can use the inject function above. The unattended upgrades question can be actually skipped more effectively. I'll add this as well.

bastianpaetzold committed:
Apply suggestions from code review

made renamed option file name more clear
made option override to prevent unattended upgrades more transparent
fixed potential shellcheck issues

Co-authored-by: MichaIng <micha@dietpi.com>

bastianpaetzold committed:
Merge branch 'dev' into pivpn_unattended_install

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Prevent unattended-upgrades dialog on interactive install

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Remove the whole reboot dialog from the installer instead of just ignoring the choice
- DietPi-Software | PiVPN: Add changelog entry and documentation in dietpi.txt about the unattended install feature

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Add missing dependency and remove reboot dialogue on uninstall

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Tiny

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Tiny

MichaIng committed:
v8.6

- DietPi-Software | PiVPN: Tiny
